Issue 36937: Add links to raw data in the targetedms replicates view

vagisha opened the pull request:
#### Rationale
Add links in the SampleFileTable grid to download raw data if it was uploaded.

#### Changes
- Added a "File" column to SampleFileTable that displays the file name extracted from SampleFile.FilePath
- Added a "Download" column with a link and download size
- Displaying links in the SampleFileTable only requires a matching row in the exp.data table.  Zip files (e.g. raw.zip for Waters data) are not validated for expected contents
- MsDataSource also has methods for validating zip files which will be used by the panoramapublic module.
- Tests added and updated; Dummy raw data added for tests

vagisha committed:
Added filename column. working on adding raw data link

vagisha committed:
First version

vagisha committed:
Working MsDataSourceUtil

vagisha committed:
Added DataSource table.

vagisha committed:
First version.

vagisha committed:
Added some tests

vagisha committed:
Added tests

vagisha committed:
Updated tests; added test files.

vagisha committed:
Cleanup

vagisha committed:
Updated error messages in unit tests.

vagisha committed:
Issue 36937: Add links to raw data in the targetedms replicates view

vagisha committed:
minor fixes in comments.

labkey-jeckels commented on src/org/labkey/targetedms/MsDataSourceUtil.java:
```diff
@@ -0,0 +1,1243 @@
+
+import static org.hamcrest.CoreMatchers.hasItem;
+
+public class MsDataSourceUtil
```
This a big class. I recommend creating a new package under org.labkey.targetedms (.raw, maybe), moving this into it, and pulling all of the nested classes into their own top-level classes.

vagisha replied:
Done. Moved to org.labkey.targetedms.datasource

labkey-jeckels commented on src/org/labkey/targetedms/MsDataSourceUtil.java:
```diff
@@ -0,0 +1,1243 @@
+                }
+                catch (IOException e)
+                {
+                    LOG.debug("Error getting size of " + dataPath, e);
```
IOExceptions indicate there's something going wrong with the file system. debug() level messages will be effectively ignored via the logging configuration by default. I recommend throwing an UnexpectedException instead, wrapping the original IOException.

If there are specific cases where you want the error to be non-fatal, please log at least at the warn level so that it's more likely an admin will see it and correct the problem

vagisha replied:
Done.

labkey-jeckels commented on src/org/labkey/targetedms/TargetedMSServiceImpl.java:
```diff
@@ -228,4 +237,10 @@ public Integer importSkylineDocument(ViewBackgroundInfo info, Path skylinePath)
     }
+
+    @Override
+    public List<? extends ISampleFile> hasData(List<? extends ISampleFile> sampleFiles, Container container)
```
I'd expect a "has" method to return true/false. Make this getData() or getSampleFiles() instead?

vagisha replied:
I changed it to getSampleFilesWithData(). Hope that is OK.

labkey-jeckels commented on src/org/labkey/targetedms/query/SampleFileTable.java:
```diff
@@ -171,4 +205,80 @@ public QueryUpdateService getUpdateService()
+                .append(" ELSE ").append(filePathColSql).append(" END");
+    }
+
+    private class DownloadLinkColumn extends DataColumn
```
Should override isSortable() and isFilterable(), since they'll be operating on the underlying column value (a long)

vagisha replied:
Done. Added isSortable() and isFilterable() overrides for the download column of the TargetedMS Runs table as well.

vagisha committed:
code review feedback:
- Moved MsDataSourceUtil to datasource package and moved nested classes to their own top-level classes
- Wrap IOException in UnexpectedException
- Changed hasData to getSampleFilesWithData
- isSortable() and isFilterable() for data download column

Updated psi-ms-PARSED.XML to include new instrument terms added to PSI-MS ontology
